tests/test_sqlRegistry.py

```diff
@@ -189,7 +189,7 @@ def testStorageInfo(self):
     def testAssembler(self):
         registry = Registry.fromConfig(self.configFile)
         storageClass = makeNewStorageClass("testAssembler")()
-        datasetType = DatasetType(name="test", dataUnits=("camera",), storageClass=storageClass)
+        datasetType = DatasetType(name="test", dataUnits=("Camera",), storageClass=storageClass)
```

Comment:
`Camera` here and `camera` a few lines later so maybe they are case insensitive?

Reply:
Note that these are slightly different contexts; this one probably should be case-sensitive and require upper/camel case, since it's definitely a DataUnit type, not a field in a DataUnit table.  The one a few lines later is where I think we ought to be case insensitive (as discussed in the comment at https://github.com/lsst/daf_butler/pull/34/files#diff-f06c7b82277ef9b866c66ef83c24c846R82), but probably do so by translating user inputs to lower/snake case.